### PR TITLE
Avoid frequent repo outages with retry

### LINKED ISSUE #11: Add a "retry" function to linux userdata and use it with yum install
yum repos are frequently temporarily unavailable, and yum has no automatic retry. To make the yum install more robust, use a retry function.

Errors:

```
Setting up Install Process
Error: Cannot retrieve repository metadata (repomd.xml) for repository: rhui-REGION-client-config-server-6. Please verify its path and try again
Could not retrieve mirrorlist https://rhui2-cds01.us-east-1.aws.ce.redhat.com/pulp/mirror/rhui-client-config/rhel/server/6/x86_64/os error was
12: Timeout on https://rhui2-cds01.us-east-1.aws.ce.redhat.com/pulp/mirror/rhui-client-config/rhel/server/6/x86_64/os: (28, 'connect() timed out!')
```

Possible retry function:

```
retry()
{
    local n=0
    local try=$1
    local cmd="${*: 2}"
    local result=1
    [[ $# -le 1 ]] && {
        echo "Usage $0 <number_of_retry_attempts> <Command>"
        exit $result
    }

    echo "Will try $try time(s) :: $cmd"

    if [[ "${SHELLOPTS}" == *":errexit:"* ]]
    then
        set +e
        local ERREXIT=1
    fi

    until [[ $n -ge $try ]]
    do
        sleep $n
        $cmd
        result=$?
        if [[ $result -eq 0 ]]
        then
            break
        else
            ((n++))
            echo "Attempt $n, command failed :: $cmd"
        fi
    done

    if [[ "${ERREXIT}" == "1" ]]
    then
        set -e
    fi

    return $result
}  # ----------  end of function retry  ----------
```

Usage:

```
retry 5 yum install git
```

### PR DESCRIPTION
**Pull request by YakDriver**
Fixes #11

### CODE REVIEW

**Comment by YakDriver** [comment]
Fixed build id, added a comment about the build id, and squashed the extraneous commits.

File: buildspec.yml [hunk 1]
```diff
@@ -77,9 +77,11 @@ phases:
       - export TF_VAR_tfi_lx_userdata_log="${TFI_LX_USERDATA_LOG}"
       - export TF_VAR_tfi_s3_bucket="${TFI_S3_BUCKET}"
       - export TF_VAR_tfi_build_date=$(date +'%Y%m%d')
-      - export TF_VAR_tfi_build_id=$(date +'%H%M')"_"$(echo $CODEBUILD_BUILD_ID | cut -d'-' -f 5)
-      - #export TF_VAR_tfi_build_id=$(echo $CODEBUILD_BUILD_ID | cut -d'-' -f 5)
-      - export TF_VAR_tfi_instance_name_tag="terrafirm-"$(echo $CODEBUILD_BUILD_ID | cut -d'-' -f 5)
+      - # real cb env looks like "<build project>:fb1b0f16-3863-4cdd-9251-4b3efe3d6796" - here using last chunk (eg, 4b3efe3d6796) for shortness
```
Comment (lorengordon) [inline buildspec.yml]: `fb1b0f16-3863-4cdd-9251-4b3efe3d6796` is a uuid, let's shorten the helper to `"<build project>:<build uuid>"`

File: buildspec.yml [hunk 2]
```diff
@@ -77,9 +77,11 @@ phases:
...
+      - build_id=$(echo $CODEBUILD_BUILD_ID | cut -d':' -f 2 | cut -d'-' -f 5)
+      - export TF_VAR_tfi_build_id=$(date +'%H%M')"_"$build_id
+      - #export TF_VAR_tfi_build_id=$build_id
```
Comment (lorengordon) [inline buildspec.yml]: let's remove this extraneous comment

**Comment by YakDriver** [comment]
Changes made

**Pull request merged**